libminifi/src/provenance/ProvenanceRepository.cpp

```diff
@@ -36,6 +36,7 @@ void ProvenanceRepository::run() {
     uint64_t curTime = getTimeMillis();
     uint64_t size = repoSize();
     if (size >= purgeThreshold) {
+      std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
If LevelDB is thread safe and is being picked up that means at that moment in the thread which picks it it is not eligible to be purged and is eligible to be sent. That means you can atomically remove said item and thread A will never get it to send OR lock all access. If you need locking here above LevelDB we should change the access pattern.